### PR TITLE
Support operating on entities and groups by their names

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

File: vault/identity_store_groups.go [hunk 1]
```diff
@@ -346,124 +378,160 @@ func (i *IdentityStore) pathGroupIDDelete() framework.OperationFunc {
...
+	if group == nil {
+		return nil, nil
+	}
+
+	ns, err := namespace.FromContext(ctx)
+	if err != nil {
+		return nil, err
+	}
+	if group.NamespaceID != ns.ID {
+		return nil, logical.ErrUnsupportedPath
```
Comment (briankassouf) [inline vault/identity_store_groups.go]: I think we should `return nil, nil` here so it matches the case where there is no group

File: vault/identity_store_entities.go [hunk 2]
```diff
@@ -423,78 +451,151 @@ func (i *IdentityStore) pathEntityIDDelete() framework.OperationFunc {
...
+		if entity == nil {
+			return nil, nil
+		}
+
 		ns, err := namespace.FromContext(ctx)
 		if err != nil {
 			return nil, err
 		}
+		if entity.NamespaceID != ns.ID {
+			return nil, logical.ErrUnsupportedPath
```
Comment (briankassouf) [inline vault/identity_store_entities.go]: Same we should `return nil, nil` here

File: vault/identity_store_groups.go (hunk 1 above)
Comment (vishalnayak) [reply vault/identity_store_groups.go]: Fixed

File: vault/identity_store_entities.go (hunk 2 above)
Comment (vishalnayak) [reply vault/identity_store_entities.go]: Fixed

**Pull request merged**